crates/interpreter/src/instructions/memory.rs

```diff
@@ -35,3 +42,35 @@ pub fn msize(interpreter: &mut Interpreter, _host: &mut dyn Host) {
     gas!(interpreter, gas::BASE);
     push!(interpreter, U256::from(interpreter.memory.effective_len()));
 }
+// From EIP-5656 MCOPY
+pub fn mcopy<SPEC: Spec>(interpreter: &mut Interpreter, host: &mut dyn Host) {
+    check!(interpreter, SPEC::enabled(CANCUN));
+    pop!(interpreter, dest, src, len);
+    if len == U256::ZERO {
```

Comment:
This is not correct, even if len is zero, some gas needs to be deduced.
maybe it is best to check if `len == 0` after `[gas_or_fail](gas::verylowcopy_cost)` is added.